src/main/java/uk/gov/justice/hmpps/casenotes/controllers/AdminController.kt

```diff
@@ -0,0 +1,108 @@
+package uk.gov.justice.hmpps.casenotes.controllers
+
+import io.swagger.v3.oas.annotations.Operation
+import io.swagger.v3.oas.annotations.media.Content
+import io.swagger.v3.oas.annotations.media.Schema
+import io.swagger.v3.oas.annotations.responses.ApiResponse
+import io.swagger.v3.oas.annotations.responses.ApiResponses
+import io.swagger.v3.oas.annotations.tags.Tag
+import jakarta.validation.Valid
+import org.springframework.http.HttpStatus
+import org.springframework.security.access.prepost.PreAuthorize
+import org.springframework.web.bind.annotation.DeleteMapping
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.PutMapping
+import org.springframework.web.bind.annotation.RequestBody
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.ResponseStatus
+import org.springframework.web.bind.annotation.RestController
+import uk.gov.justice.hmpps.casenotes.config.ADMIN_ONLY
+import uk.gov.justice.hmpps.casenotes.config.CaseNoteRequestContext
+import uk.gov.justice.hmpps.casenotes.config.SecurityUserContext.Companion.ROLE_CASE_NOTES_ADMIN
+import uk.gov.justice.hmpps.casenotes.legacy.dto.ErrorResponse
+import uk.gov.justice.hmpps.casenotes.notes.CaseNote
+import uk.gov.justice.hmpps.casenotes.notes.CaseNoteAdminService
+import uk.gov.justice.hmpps.casenotes.notes.DeleteCaseNoteRequest
+import uk.gov.justice.hmpps.casenotes.notes.ReplaceNoteRequest
+import java.util.UUID
+
+@Tag(name = ADMIN_ONLY)
+@RestController
+@RequestMapping("admin")
+@PreAuthorize("hasRole('$ROLE_CASE_NOTES_ADMIN')")
+class AdminController(private val caseNoteAdmin: CaseNoteAdminService) {
+
+  @Operation(summary = "Endpoint to replace a case note and amendments.")
+  @ApiResponses(
+    value = [
+      ApiResponse(
+        responseCode = "200",
+        description = "Case Note successfully replaced",
+      ),
+      ApiResponse(
+        responseCode = "400",
+        description = "Bad request",
+        content = [Content(schema = Schema(implementation = ErrorResponse::class))],
+      ),
+      ApiResponse(
+        responseCode = "401",
+        description = "Unauthorised, requires a valid Oauth2 token",
+        content = [Content(schema = Schema(implementation = ErrorResponse::class))],
+      ),
+      ApiResponse(
+        responseCode = "403",
+        description = "Forbidden, requires an appropriate role",
+        content = [Content(schema = Schema(implementation = ErrorResponse::class))],
+      ),
+      ApiResponse(
+        responseCode = "404",
+        description = "Case note not found",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))],
+      ),
+    ],
+  )
+  @PutMapping("/case-notes/{personIdentifier}/{id}")
```

Comment:
Realised late that we don't really need the person identifier in the URL. It follows the pattern however the pattern is no longer the recommended approach. No unnecessary identifiers in paths. It's there for the GET because it was proxying to a Prison API endpoint that needed it. I may add a small amount of extra checks for the client as the need to have the id and who it belongs to but these are admin endpoints. Would be of more value if the ids were still numerical sequence based to prevent accidental or deliberate incorrect usage. Removing them would slightly simplify the code removing a bit of guard code and maybe reduce the caller responsibility a little.

Not a strongly held opinion though so feel free to argue against it!

Reply:
Yeah it's the get note and add amendment endpoints that have it and don't need it any more. We can't change those. Sync doesn't have it and these are closest conceptually to those endpoints. I say remove it